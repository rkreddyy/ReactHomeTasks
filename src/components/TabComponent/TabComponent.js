import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './TabComponent.css';

export default function TabComponent(props) {
    const [activeTab, setActiveTab] = useState(0);
    const onTabChange = (index) => {
        setActiveTab(index);
        props.tabChanged(index);
    };

    return (
        <div className="tab-container">
            {props.tabs.map((value, index) => {
                return (
                    <div
                        key={index}
                        onClick={() => onTabChange(index)}
                        className={classnames('tab', {
                            'active': index === activeTab
                        })}
                    >
                        {value}
                    </div>
                );
            })}
        </div>
    );
}

TabComponent.propTypes = {
  tabs: PropTypes.array,
  tabChanged: PropTypes.func
};
