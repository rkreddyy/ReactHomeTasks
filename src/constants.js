export const TABS = ["All", "Documentory", "Comedy", "Horror", "Crime"];