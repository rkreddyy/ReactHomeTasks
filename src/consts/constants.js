export const TABS = ['All', 'Documentory', 'Comedy', 'Horror', 'Crime'];
export const GENRE = ['Horror', 'Comedy', 'Crime', 'Documentory'];
export const DEFAULT_SORT_BY = 'release_date';
export const MENU = ['edit', 'delete'];
export const DEFAULT_FILTER_BY = 'All';
