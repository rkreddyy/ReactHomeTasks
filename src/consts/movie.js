export const MOVIE = {
    title: '',
    releaseDate: '',
    movieURL: '',
    genre: '',
    overview: '',
    runtime: '',
    src:'/public/avengers.PNG',
    category:'Action & Adventure'
};
