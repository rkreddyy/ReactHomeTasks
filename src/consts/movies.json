[
    {
      "title": "Pulp Fiction",
      "src": "/public/pulp-fiction.PNG",
      "releaseDate": "2004-10-30",
      "category": "Action & Adventure",
      "movieURL": "movie.com",
      "genre": "Comedy",
      "overview": "Pulp Fiction is a 1994 American crime filmnwritten  and directed by Quentin Tarantino; it is based on a story by Tantino and Roger Avary.Starring John Travolta,Samuel L.Jackson,Bruce Wills, Tim Roth and Uma Thurman, It tells several stories of criminal Los Angeles",
      "runtime": "154",
      "id":0
    },
    {
      "title": "Inception",
      "src": "/public/inception.png",
      "releaseDate": "1994-10-30",
      "category": "Action & Adventure",
      "movieURL": "movie.com",
      "genre": "Comedy",
      "overview": "overview",
      "runtime": "154",
      "id":1
    },
    {
      "title": "Bohemian Rhapsody",
      "src": "/public/bohemian.PNG",
      "releaseDate": "2004-10-30",
      "category": "Drama, Biography, Music",
      "movieURL": "movie.com",
      "genre": "Comedy",
      "overview": "overview",
      "runtime": "154",
      "id":2
    },
    {
      "title": "Kill the bill",
      "src": "/public/killthebill.PNG",
      "releaseDate": "1994-10-30",
      "category": "Oscar Winning Movie",
      "movieURL": "movie.com",
      "genre": "Comedy",
      "overview": "overview",
      "runtime": "154",
      "id":3
    },
    {
      "title": "Bohemian Rhapsody",
      "src": "/public/bohemian.PNG",
      "releaseDate": "2004-10-30",
      "category": "Drama, Biography, Music",
      "movieURL": "movie.com",
      "genre": "Horror",
      "overview": "overview",
      "runtime": "154",
      "id":4
    },
    {
      "title": "Inception",
      "src": "/public/inception.png",
      "releaseDate": "2003-10-30",
      "category": "Action & Adventure",
      "movieURL": "movie.com",
      "genre": "Horror",
      "overview": "overview",
      "runtime": "154",
      "id":5
    },
    {
      "title": "Pulp Fiction",
      "src": "/public/pulp-fiction.PNG",
      "releaseDate": "1994-10-30",
      "category": "Action & Adventure",
      "movieURL": "movie.com",
      "genre": "Crime",
      "overview": "Pulp Fiction is a 1994 American crime filmnwritten  and directed by Quentin Tarantino; it is based on a story by Tantino and Roger Avary.Starring John Travolta,Samuel L.Jackson,Bruce Wills, Tim Roth and Uma Thurman, It tells several stories of criminal Los Angeles",
      "runtime": "154",
      "id":6
    },
    {
      "title": "Bohemian Rhapsody",
      "src": "/public/bohemian.PNG",
      "releaseDate": "2004-10-30",
      "category": "Drama, Biography, Music",
      "movieURL": "movie.com",
      "genre": "Crime",
      "overview": "Bohemian Rhapsody is a 2004 American crime filmnwritten  and directed by Quentin Tarantino; it is based on a story by Tantino and Roger Avary.Starring John Travolta,Samuel L.Jackson,Bruce Wills, Tim Roth and Uma Thurman, It tells several stories of criminal Los Angeles",
      "runtime": "154",
      "id":7
    }
  ]