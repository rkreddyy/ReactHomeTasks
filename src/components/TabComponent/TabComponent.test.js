import TabComponent from './TabComponent';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {TABS} from '../../consts/constants';

describe('Tab Component', () => {
    configure({ adapter: new Adapter() });
    test('Should render Tabs', () => {   
        const component = shallow(<TabComponent tabs={TABS}></TabComponent>);
        expect(component.find('.tab-container').find('.tab').length).toEqual(TABS.length);
    });

    test('Should match the snapshot', () => {
        const component = renderer.create(<TabComponent tabs={TABS} />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
