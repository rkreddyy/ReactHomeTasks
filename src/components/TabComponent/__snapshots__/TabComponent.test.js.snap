// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Component Should match the snapshot 1`] = `
<div
  className="tab-container"
>
  <div
    className="tab active"
    onClick={[Function]}
  >
    All
  </div>
  <div
    className="tab"
    onClick={[Function]}
  >
    Documentory
  </div>
  <div
    className="tab"
    onClick={[Function]}
  >
    Comedy
  </div>
  <div
    className="tab"
    onClick={[Function]}
  >
    Horror
  </div>
  <div
    className="tab"
    onClick={[Function]}
  >
    Crime
  </div>
</div>
`;
